File: Sieve of Eratosthenes/sevie.js
//  Find all the Prime Numbers less than or equal to a given Number
//  Sieve of Eratosthenes
// The Sieve of Eratosthenes is a simple algorithm that finds the prime numbers up to a given integer.
function sieve(n){
    let prime=new Array(n+1).fill(true)
    prime[0]=false
    prime[1]=false
    for(let i=2;i*i<=n;i++){
        if(prime[i]){
            for(let j=i*i;j<=n;j=j+i){
                prime[j]=false
            }
        }
    }
    for(let i=0;i<=n;i++){
        if(prime[i]){
            console.log(i);
        }
    }
}

sieve(100)

File: Sieve of Eratosthenes/sevie.py
# Sieve of Eratosthenes algorithm
# Created to find prime numbers upto a given number
# Time complexity: O(n*log(log(n)))

def sieve(n):
    prime = [True for i in range(n+1)]
    p = 2
    while p*p <= n:
        if prime[p] == True:
            for i in range(p*p,n+1,p):
                prime[i] = False
        p += 1
    for p in range(2,n+1):
        if prime[p]:
            print(p,end=" ")
    print()

# Example
sieve(10)